src/integer/vec_z/from.rs

```diff
@@ -0,0 +1,263 @@
+            }),
+        }
+    }
+}
+
+impl FromStr for VecZ {
+    type Err = MathError;
+
+    /// Creates a [`VecZ`] row-vector with entries in [`Z`] from a [`String`].
+    /// The format of that string looks like this `[1,2,3]` for a row vector
+    /// with three entries (`1` in the first row, `2` in the second one, ...)
+    ///
+    /// Parameters:
+    /// - `string`: the vector representation as a string
+    ///
+    /// Returns a [`VecZ`] or an error, if the vector is not formatted in a suitable way,
+    /// the number of rows is too big (must fit into [`i64`]), or if the regular expression
+    /// inside of the function could not be processed.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::VecZ;
+    /// use std::str::FromStr;
+    ///
+    /// let string = String::from("[[1, 2, 3]]");
+    /// let matrix = VecZ::from_str(&string).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`InvalidMatrix`](MathError::InvalidMatrix)
+    /// if the vector is not formatted in a suitable way,
+    /// the number of rows is too big (must fit into [`i64`]).
+    /// - Returns a [`MathError`] of type
+    /// [`InvalidStringToCStringInput`](MathError::InvalidStringToCStringInput)
+    /// if an entry contains a Nul byte.
+    /// - Returns a [`MathError`] of type
+    /// [`InvalidStringToZInput`](MathError::InvalidStringToZInput)
+    /// if an entry is not formatted correctly.
+    fn from_str(string: &str) -> Result<Self, MathError> {
+        let entries_string = parse_matrix_string(string)?;
+        let (num_rows, num_cols) = find_matrix_dimensions(&entries_string)?;
+        let mut vector: VecZ;
+        match (num_rows, num_cols) {
+            (1, _) => vector = VecZ::new(num_cols, VectorDirection::RowVector)?,
+            (_, 1) => vector = VecZ::new(num_rows, VectorDirection::ColumnVector)?,
+            (_, _) => return Err(MathError::InvalidStringToVectorInput(String::from(
+                "The string either contained more than one column or row, or did have zero entries",
+            ))),
+        }
+
+        // fill entries of matrix according to entries in string_matrix
+        for (row_num, row) in entries_string.iter().enumerate() {
+            for (col_num, entry) in row.iter().enumerate() {
+                let z_entry = Z::from_str(entry)?;
+                vector.matrix.set_entry(row_num, col_num, z_entry)?;
+            }
+        }
+        Ok(vector)
+    }
+}
+
+#[cfg(test)]
+mod test_new {
+
+    use crate::integer::{VecZ, Z};
+    use crate::utils::VectorDirection;
+
+    /// Ensure that entries of a new vector are `0`.
+    #[test]
+    fn entry_zero() {
+        let matrix = VecZ::new(2, VectorDirection::ColumnVector).unwrap();
+
+        let entry1 = matrix.get_entry(0).unwrap();
+        let entry2 = matrix.get_entry(0).unwrap();
+
+        assert_eq!(Z::from_i64(0), entry1);
+        assert_eq!(Z::from_i64(0), entry2);
```

Comment:
though i prefer using the constant
```suggestion
        assert_eq!(Z::from(0), entry1);
        assert_eq!(Z::from(0), entry2);
```

Reply:
Changed to Z::ZERO